week-03/assessment/src/main/java/solarfarm/ui/Controller.java

```diff
@@ -0,0 +1,115 @@
+package solarfarm.ui;
+
+import solarfarm.data.DataException;
+import solarfarm.domain.SolarFarmResult;
+import solarfarm.domain.SolarFarmService;
+import solarfarm.models.SolarPanel;
+
+import java.util.List;
+
+
+public class Controller {
+    private final SolarFarmService service;
+    private final View view;
+
+    public Controller(SolarFarmService service, View view) {
+        this.service = service;
+        this.view = view;
+    }
+    public void run() { // this is where all those data exceptions are handled
+        view.printHeader("Welcome to the SolarFarm");
+        try{
+            runApp();
+        }
+        catch (DataException DE){
+            view.displayErrors(List.of(DE.getMessage()));
+        }
+    }
+
+    private void runApp() throws DataException{
+        boolean exit = false;
+        while(!exit){
+            int choice = view.displayMenu(); //Menu options
+            switch (choice) {
+                case 0 -> exit = true;
+                case 1 -> viewAll();
+                case 2 -> viewBySection();
+                case 3 -> addPanel();
+                case 4 -> updatePanel();
+                case 5 -> findBySRC();
+                case 6 -> removeBySRC();
```

Comment:
Minor: technically this class is through Java 11 which does not have this switch syntax